### PR TITLE
Put in the basic jujulib.js skeleton

### PR DESCRIPTION
**Pull request by jaycee**

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -14,6 +14,9 @@ module.exports = function(config) {
 
     // list of files / patterns to load in the browser
     files: [
+      'jujugui/static/gui/build/app/assets/javascripts/jujulib/juju.js',
```
Comment (hatched) [inline karma.conf.js]: these two lines can be shortened to `'jujugui/static/gui/build/app/assets/javascripts/jujulib/*.js'`

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2962/
Test PASSed.

File: jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*
+Copyright (C) 2015 Canonical Ltd.
+
+XXX jcssackett 2015-09-18: Licensing for juju.js? It's different then the
+licensing for the GUI.
+*/
+
+;
+
+var jujulib = (function () {
```
Comment (hatched) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js]: To make this work in both nodejs and the browser you can use this technique:

```JavaScript
(function(exports){

   exports.charmstore = function() {};
   exports.environment = function() {};
   exports.identity = function() {};

})(typeof exports === 'undefined'? this.jujulib={}: exports);
```

File: karma.conf.js (hunk 1 above)
Comment (jaycee) [inline karma.conf.js]: Cool, done.

File: jujugui/static/gui/src/app/assets/javascripts/jujulib/test-juju.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+chai.config.includeStack = true;
+chai.config.truncateThreshold = 0;
+
+describe('juju.js', function() {
+
+    it('exists', function() {
+        var cs = new jujulib.charmstore();
+        var env = new jujulib.environment();
+        var identity = new jujulib.identity();
+        assert.isObject(cs);
```
Comment (hatched) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/test-juju.js]: I suppose you'd probably want to do:

```JavaScript
assert.isTrue(cs instanceof jujulib.charmstore);
```

Comment (jaycee) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/test-juju.js]: That's a good idea.

File: jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js (hunk 2 above)
Comment (jaycee) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js]: Cool thanks.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2963/
Test PASSed.

**Comment by hatched** [comment]
:+1:

**Comment by jaycee** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2964/
Test PASSed.

**Pull request merged**